src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Azure.WebJobs.Script.Configuration;
+using Microsoft.Azure.WebJobs.Script.Properties;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace Microsoft.Azure.WebJobs.Script.ExtensionBundle
+{
+    public class ExtensionBundleManager : IExtensionBundleManager
+    {
+        private readonly IEnvironment _environment;
+        private readonly ExtensionBundleOptions _options;
+        private readonly ILogger _logger;
+        private readonly string _cdnUri;
+
+        public ExtensionBundleManager(ExtensionBundleOptions options, IEnvironment environment, ILoggerFactory loggerFactory)
+        {
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryExtensionBundles) ?? throw new ArgumentNullException(nameof(loggerFactory));
+            _cdnUri = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AlternateCdnUri) ?? ScriptConstants.CdnBaseUri;
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public bool IsExtensionBundleConfigured()
+        {
+            return !string.IsNullOrEmpty(_options.Id) && !string.IsNullOrEmpty(_options.Version?.OriginalString);
+        }
+
+        public async Task<string> GetExtensionBundle(HttpClient httpClient = null)
+        {
+            if (httpClient == null)
+            {
+                using (httpClient = new HttpClient())
+                {
+                    return await GetBundle(httpClient);
+                }
+            }
+            else
+            {
+                return await GetBundle(httpClient);
+            }
```

Comment:
Remove Method GetBundle. Move logic to new up HttpClient if null to GetExtensionBundle.

Reply:
can't new up the HttpClient in GetBundle method. The httpClient is inside using statement in one instance vs the other one is plainly using it. Removing GetBundle method. I would have to copy the entire logic inside if block as well as the else block statement.

Reply:
To have a simpler API,  you should have two distinct overloads of GetExtensionBundle, the one that does not take an `HttpClient` would dispose of the client it creates.